sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_properties.asciidoc

```diff
@@ -0,0 +1,175 @@
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+- link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties]
+- link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr]
+
+
+== Proposal
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_USM_PROPERTIES` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+=== Examples
+
+The following is a synopsis of the new USM malloc API.
+
+[source,c++]
+----
+
+// A property list containing a compile-time property and a runtime property
+sycl::ext::oneapi::experimental::properties properties{some_compile_time_prop<1>, some_runtime_prop(1)};
+
+// The compile time property is passed to template arguments of annotated_ptr
+auto data = sycl::ext::oneapi::experimental::malloc_device(N, q, properties);
+// data is of type annotated_ptr<void*, property_list_t<some_compile_time_prop<1>>>
+
+auto data2 = sycl::ext::oneapi::experimental::malloc_device<int>(N, q, properties);
+// data2 is of type annotated_ptr<int*, property_list_t<some_compile_time_prop<1>>>
+
+sycl::queue q;
+q.parallel_for(range<1>(N), [=] (id<1> i){
+  data2[i] *= 2;
+}).wait();
+----
+
+
+=== New USM allocation APIs
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Available only when is_property_list<PropertyListT>::value is true
+template <typename T = void*, typename PropertyListT = property_list_t<>>
+annotated_ptr<T, PropertyListT> malloc_device(
+    size_t Count, const queue &Q, PropertyListT &PropList) {
+  ...
+}
+
+// Available only when is_property_list<PropertyListT>::value is true
+template <typename T = void*, typename PropertyListT = property_list_t<>>
+annotated_ptr<T, PropertyListT> malloc_shared(
+    size_t Count, const queue &Q, PropertyListT &PropList) {
+  ...
+}
+
+// Available only when is_property_list<PropertyListT>::value is true
+template <typename T = void*, typename PropertyListT = property_list_t<>>
+annotated_ptr<T, PropertyListT> malloc_host(
+    size_t Count, const context &Ctxt, PropertyListT &PropList) {
+  ...
+}
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+The same setup is applied to other overloads of malloc APIs.
+
+Compile time properties can be pass into runtime properties within the allocation function.
```

Comment:
What does this  mean?

Reply:
I have removed this sentence given it is more of an implementation detail specific to subset of the properties.